src/test/scala/utils/KafkaTestContainer.scala

```diff
@@ -0,0 +1,63 @@
+package utils
+
+import cats.effect.syntax.all.*
+import cats.effect.{Ref, Resource, Sync}
+import cats.syntax.all.*
+import org.scalatest.Assertion
+import org.testcontainers.containers.KafkaContainer as Underlying
+import org.testcontainers.utility.DockerImageName
+import utils.KafkaContainer.KafkaConfig
+
+trait KafkaTestContainer[F[_]] {
+  def withKafkaContext(test: KafkaConfig => F[Assertion])(using Sync[F]): F[Assertion] =
```

Comment:
i dont find this name very clear, i think `withRunningKafka` is clearer

Reply:
https://github.com/sky-uk/fs2-kafka-topic-loader/pull/105/commits/d5f8271fff8442b76fe9f4350f2b3aabf7512450